avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/FeatureIndicesConversion.scala

```diff
@@ -181,18 +208,27 @@ object FeatureIndicesConversion {
    * Spark UDF function to convert a column of String Seq to a column of Id Seq
    *
    * @param featureMapping The mapping of word to id
+   * @param discardUnknownEntries Whether to discard unknown entries
    * @return A Spark udf
    */
-  private def convertStringSeqToIdSeq(featureMapping: Map[String, Long]): UserDefinedFunction = {
+  private def convertStringSeqToIdSeq(
+    featureMapping: Map[String, Long],
+    discardUnknownEntries: Boolean): UserDefinedFunction = {
 
     udf {
       stringSeq: Seq[String] => {
-        val cardinality = featureMapping.size.toLong
+        val lastIndex = if (discardUnknownEntries) featureMapping.size.toLong - 1 else featureMapping.size.toLong
```

Comment:
does the featureMapping include "unknown" or not?

Reply:
no feature mapping does not include unknown